### PR TITLE
Upgrade core dependencies

### PR DESCRIPTION
**Pull request by kingsleyadio**
Upgrades core dependencies
- **Gradle** 5.1 => 5.2
- **AGP** 3.4.0-alpha10 => 3.4.0-rc01
- **Kotlin** 1.3.11 => 1.3.21

### CODE REVIEW

File: gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -1,5 +1,5 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-5.1-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-5.2-all.zip
```
Comment (eyedol) [inline gradle/wrapper/gradle-wrapper.properties]: should we get the bin version here as well?

File: deeplink/src/test/java/com/hellofresh/deeplink/DeepLinkUriTest.kt [hunk 2]
```diff
@@ -1833,14 +1833,11 @@ class DeepLinkUriTest {
...
-        @Suppress("unused")
+        @Suppress("unused", "BooleanLiteralArgument")
         @Parameterized.Parameters(name = "Use get = {0}")
         @JvmStatic
-        fun parameters(): Collection<Array<Any>> {
-            return Arrays.asList(
-                arrayOf<Any>(true),
-                arrayOf<Any>(false)
-            )
+        fun parameters(): Collection<*> {
```
Comment (eyedol) [inline deeplink/src/test/java/com/hellofresh/deeplink/DeepLinkUriTest.kt]: Would have been nice if this was done a  separate commit.

Comment (kingsleyadio) [reply deeplink/src/test/java/com/hellofresh/deeplink/DeepLinkUriTest.kt]: Aha. I can extract it into it's own commit then

File: gradle/wrapper/gradle-wrapper.properties (hunk 1 above)
Comment (kingsleyadio) [reply gradle/wrapper/gradle-wrapper.properties]: bin is the one without source IIRC. `all` comes with everything bundled

Comment (eyedol) [reply gradle/wrapper/gradle-wrapper.properties]: ignore, I meant the sources.
